freesteel/card.py

```diff
@@ -126,3 +127,74 @@ def disconnect(self):
 			# failed selecting path
 			message = scard.SCardGetErrorMessage(result)
 			raise DisconnectCardError('Error disconnecting card: {0}'.format(message))
+
+
+class GemaltoCard(AltoCard):
+	CARD_AID = [
+		0xF3, 0x81, 0x00, 0x00, 0x02, 0x53, 0x45,
+		0x52, 0x49, 0x44, 0x01
+	]
+
+	def __init__(self, handle, protocol):
+		self._handle = handle
+		self._protocol = protocol
+
+		command = [0x00, 0xA4, 0x04, 0x00]
+		command.append(len(self.CARD_AID))
+		command.extend(self.CARD_AID)
+		command.append(0x00)
+		scard.SCardTransmit(
+			handle,
+			protocol,
+			command
+		)
+
+	def select_path(self, path, ne=4):
+		"""Select path on card"""
+		request = [0x00, 0xA4, 0x08, 0x00]
+		request.append(len(path))
+		request.extend(path)
+		request.append(ne)
+
+		# transmit command to card
+		result, response = scard.SCardTransmit(
+			self._handle,
+			self._protocol,
+			request
+		)
+
+		# raise error if needed
+		if result != scard.SCARD_S_SUCCESS:
+			# failed selecting path
+			message = scard.SCardGetErrorMessage(result)
+			raise SelectPathError('Error selecting path: {0}'.format(message))
+
+		return response
+
+	def read_file(self, card_file, strip_tag=False):
+		"""Select and read file from card"""
+		file_info = self.select_path(card_file, 4)
+
+		length = (file_info[2] << 8) + file_info[3]
+		offset = 0
+		known_real_length = False
+
+		output = []
+		while length > 0:
+			data = self.read_binary(offset, length)
+			if not known_real_length:
+				# get length from outher tag, skip first 4 bytes (outher tag + length)
+				# if strip_tag is true, skip 4 more bytes (inner tag + length) and return content
```

Comment:
To ne znaci ni da je onaj drugi komentar koristan. Opisuje sta kod radi komandu po komandu, ne zasto to radi.

To je kao kad bih ja imao:
```python
# increases x by one
x += 1
```
Umesto:

```python
# compensate for border
x += 1
```
No, pretpostavljam da je kod samo prepisan odande pa cu mergovati svakako.